lib/mail/message.rb

```diff
@@ -356,11 +356,11 @@ def <=>(other)
     #  m1 == m2 #=> false
     def ==(other)
       return false unless other.respond_to?(:encoded)
-
       if self.message_id && other.message_id
         self.encoded == other.encoded
       else
-        self_message_id, other_message_id = self.message_id, other.message_id
+        self_message_id = header[:message_id] ? header[:message_id].value : nil
```

Comment:
Given a message id such as `<foo@bar.com>`, the existing code _was_ mutating the object by setting the parsed version directly back into the header. This pulls the value directly from the headers so that they go back the same.

This was surfaced in this bug fix as pulling attributes from the Message class no longer silently fallback to their raw form.